demos/text_detection_demo/main.cpp

```diff
@@ -60,138 +71,211 @@ int main(int argc, char *argv[]) {
+                int top_left_point_idx = 0;
+
+                if (rect.size != cv::Size2f(0, 0) && text_detection.is_initialized()) {
+                    std::chrono::steady_clock::time_point begin = std::chrono::steady_clock::now();
+                    points = floatPointsFromRotatedRect(rect);
+                    topLeftPoint(points, &top_left_point_idx);
+                    cropped_text = cropImage(image, points, text_recognition.input_size(), top_left_point_idx);
+                    std::chrono::steady_clock::time_point end = std::chrono::steady_clock::now();
+                    text_crop_time += std::chrono::duration_cast<std::chrono::microseconds>(end - begin).count();
+                } else {
+                    if (FLAGS_cc) {
+                        int w = image.cols * 0.05;
+                        int h = w * 0.5;
+                        cv::Rect r(image.cols * 0.5 - w * 0.5, image.rows * 0.5 - h * 0.5, w, h);
+                        cropped_text = image(r).clone();
+                        cv::rectangle(demo_image, r, cv::Scalar(0, 0, 255), 2);
+                        points.emplace_back(r.tl());
+                    } else {
+                        cropped_text = image;
+                    }
+                }
 
+                std::string res = "";
+                double conf = 1.0;
+                if (text_recognition.is_initialized()) {
+                    auto blobs = text_recognition.Infer(cropped_text);
+                    auto output_shape = blobs.begin()->second->getTensorDesc().getDims();
+                    if (output_shape[2] != kAlphabet.length())
+                        throw std::runtime_error("The text recognition model does not correspond to alphabet.");
 
-        /* Resize manually and copy data from the image to the input blob */
-        Blob::Ptr input = infer_request.GetBlob(input_name);
-        float* input_data = input->buffer().as<PrecisionTrait<Precision::FP32>::value_type *>();
+                    float *ouput_data_pointer = blobs.begin()->second->buffer().as<PrecisionTrait<Precision::FP32>::value_type *>();
+                    std::vector<float> output_data(ouput_data_pointer, ouput_data_pointer + output_shape[0] * output_shape[2]);
 
-        image.convertTo(image, CV_32F);
+                    std::chrono::steady_clock::time_point begin = std::chrono::steady_clock::now();
+                    res = CTCGreedyDecoder(output_data, kAlphabet, kPadSymbol, &conf);
+                    std::chrono::steady_clock::time_point end = std::chrono::steady_clock::now();
+                    text_recognition_postproc_time += std::chrono::duration_cast<std::chrono::microseconds>(end - begin).count();
 
-        cv::resize(image, image, cv::Size(input_info->getTensorDesc().getDims()[3],
-                   input_info->getTensorDesc().getDims()[2]));
+                    res = conf >= min_text_recognition_confidence ? res : "";
+                    num_found += !res.empty() ? 1 : 0;
```

Comment:
Is it better to replace on cast operation?

Reply:
I think this is not so important...

Reply:
Ok